examples/pypi_wheel_provider.py

```diff
@@ -123,19 +123,20 @@ def get_base_requirement(self, candidate):
     def get_preference(self, resolution, candidates, information):
         return len(candidates)
 
-    def find_matches(self, requirement):
+    def iter_matches(self, requirements):
         candidates = []
-        name = requirement.name
-        extras = requirement.extras
+        name = requirements[0].name
```

Comment:
Add a comment here noting that the API guarantees that name is the same for all requirements?

Reply:
I have added a line in the `AbstractProvider.iter_matches()` docstring that says *the collection is never empty*.